File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.go
func lengthOfLongestSubstring(s string) int {
    l := 0
    m := make(map[byte]bool)
    res := 0

    for r := 0 ; r < len(s) ; r++ {
        for m[s[r]] {
            delete(m, s[l])
            l++
        }
        m[s[r]] = true

        res = max(res, r-l+1)
    }
    return res

}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l = 0
        res = 0
        max_f = 0

        seen = Counter()

        for r in range(len(s)):
            seen[s[r]] += 1

            max_f = max(max_f, seen[s[r]])

            while (r-l+1) - max_f > k:
                seen[s[l]] -= 1
                l += 1
            
            res = max(res, r-l+1)
        return res